app/Http/Controllers/CreatePostingPeriod.php

```
@@ -0,0 +1,184 @@
+
+    /**
+     * Ensure posting period exists for the given month
+     */
+    private function ensureYearlyPostingPeriod(Carbon $targetDate): PostingPeriod
+    {
+        $postingPeriod = PostingPeriod::where('period_start', '<=', $targetDate)
+            ->where('period_end', '>=', $targetDate)
+            ->first();
+            
+        if (!$postingPeriod) {
+            if (PostingPeriod::scopeCheckIfStatusIsOpenYearly()) {
```

PHPMD (2.15.0)

Avoid using static access to class '\App\Models\PostingPeriod' in method 'ensureYearlyPostingPeriod'. (Clean Code Rules)

(StaticAccess)